music21/stream/makeNotation.py

```diff
@@ -840,34 +840,46 @@ def makeRests(
             )
         return returnObj
 
-    def oHighTargetForMeasure(m: stream.Measure) -> float:
-        '''Needed for timeRangeFromBarDuration'''
-        # NOTE: this returns 0.0 if no meter can be found
-        post = m.barDuration.quarterLength
-        post -= m.paddingLeft
-        post -= m.paddingRight
+    def oHighTargetForMeasure(
+        m: Optional[stream.Measure] = None,
+        ts: Optional[meter.TimeSignature] = None
+    ) -> float:
+        """
+        Needed for timeRangeFromBarDuration.
+        Returns 0.0 if no meter can be found.
+        """
+        bar_duration: float = 0.0
+        if ts is not None:
+            bar_duration = ts.barDuration
+        elif m is not None:
+            # More expensive context search
+            bar_duration = m.barDuration
+        post = bar_duration.quarterLength
```

Comment:
barDuration is typed as a float but floats don't have quarterLength.  Better as:

```
post: float = 0.0
if ts is not None:
     post = ts.barDuration.quarterLength
elif m is not None:
    # more expensive context search
    post = m.barDuration.quarterLength
```

note that the quarterLength can be a fraction (as can paddingLeft/paddingRight) so either type the func and post as Union[float, Fraction]) (best) or return float(post).

Reply:
Thanks, I missed this because I haven't tested the (unreachable, I think) error condition when both args are None.